sdk/spring/azure-identity-spring/src/main/java/com/azure/spring/identity/DefaultSpringCredentialBuilder.java

```diff
@@ -15,54 +26,52 @@
  */
 public class DefaultSpringCredentialBuilder extends SpringCredentialBuilderBase<DefaultSpringCredentialBuilder> {
 
-    /**
-     * Defines the AZURE_CREDENTIAL_PREFIX.
-     */
-    static final String AZURE_CREDENTIAL_PREFIX = "azure.credential.";
+    private String managedIdentityClientId;
+    private String tenantId;
 
-    private String alternativePrefix;
+    public DefaultSpringCredentialBuilder(Environment environment) {
+        super(environment);
+        final AzureEnvironment azureEnvironment = new AzureEnvironment(environment);
+        this.managedIdentityClientId = azureEnvironment.getClientId();
+        this.tenantId = azureEnvironment.getTenantId();
+    }
 
-    public DefaultSpringCredentialBuilder alternativePrefix(String alternative) {
-        if (alternative != null) {
-            this.alternativePrefix = alternative + (alternative.endsWith(".") ? "" : ".");
-        }
+    public DefaultSpringCredentialBuilder managedIdentityClientId(String clientId) {
+        this.managedIdentityClientId = clientId;
+        return this;
+    }
 
+    public DefaultSpringCredentialBuilder tenantId(String tenantId) {
+        this.tenantId = tenantId;
         return this;
     }
 
     /**
-     * Build a default Spring token credential, which will be a chained credential.
-     * If an alternative prefix is specified in the builder, the chain of credential
-     * will have three credentials, one with the specified prefix, one with the default
-     * spring credential prefix, and the default managed identity credential without client id
-     * set. Otherwise, the chain will consist the credential with the default prefix and the default
-     * managed identity credential.
+     * Build a {@link DefaultSpringCredential} with below list of credentials:
+     * <ol>
+     * <li>{@link SpringEnvironmentCredential}</li>
+     * <li>{@link ManagedIdentityCredential}</li>
+     * <li>{@link SharedTokenCacheCredential}</li>
+     * <li>{@link IntelliJCredential}</li>
+     * <li>{@link VisualStudioCodeCredential}</li>
+     * <li>{@link AzureCliCredential}</li>
+     * <li>{@link AzurePowerShellCredential}</li>
```

Comment:
After using this builder, we should keep the java doc twice, it's error-prone: forget to update one place when update the java doc.